src/managers/hub/SimplePriceManager.sol

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {INAVHook} from "./interfaces/INAVManager.sol";
+import {ISimplePriceManager} from "./interfaces/ISimplePriceManager.sol";
+
+import {Auth} from "../../misc/Auth.sol";
+import {D18, d18} from "../../misc/types/D18.sol";
+
+import {PoolId} from "../../common/types/PoolId.sol";
+import {IGateway} from "../../common/interfaces/IGateway.sol";
+import {ShareClassId} from "../../common/types/ShareClassId.sol";
+
+import {IHub} from "../../hub/interfaces/IHub.sol";
+import {IHubRegistry} from "../../hub/interfaces/IHubRegistry.sol";
+import {IShareClassManager} from "../../hub/interfaces/IShareClassManager.sol";
+
+/// @notice Base share price calculation manager for single share class pools.
+contract SimplePriceManager is ISimplePriceManager, Auth {
+    IGateway public gateway;
+    IHub public immutable hub;
+    IHubRegistry public immutable hubRegistry;
+    IShareClassManager public immutable shareClassManager;
+
+    mapping(PoolId poolId => Metrics) public metrics;
+    mapping(PoolId poolId => mapping(address => bool)) public manager;
+    mapping(PoolId poolId => mapping(uint16 centrifugeId => NetworkMetrics)) public networkMetrics;
+    mapping(PoolId poolId => uint16[]) internal _networks;
+
+    constructor(IHub hub_, address deployer) Auth(deployer) {
+        hub = hub_;
+        gateway = hub_.gateway();
+        hubRegistry = hub_.hubRegistry();
+        shareClassManager = hub_.shareClassManager();
+    }
+
+    modifier onlyManager(PoolId poolId) {
```

Comment:
This seems not used in `SimplePriceManager`

Reply:
ah yes, was from when it had manager actions. removed